src/wasm-binary.h

```diff
@@ -445,15 +469,22 @@ class WasmBinaryWriter : public WasmVisitor<void> {
           << uint16_t(numLocalsByType[f64]);
       }
       if (function) {
-        size_t curr = o.size();
-        visit(function->body);
-        o.writeAt(curr, uint16_t(o.size() - curr));
+        size_t sizePos = o.size();
+        o << (uint16_t)0; // placeholder
+        size_t start = o.size();
+        depth = 0;
+        recurse(function->body);
+        assert(depth == 0);
+        size_t size = o.size() - start;
+        assert(size <= std::numeric_limits<uint16_t>::max());
```

Comment:
v8 binary format document says so :) yes, definitely too small for real-world code.